### PR TITLE
LOF-22 Create alpine-based micros base image for new Experience env

### PR DESCRIPTION
**Pull request by matteoluzzi**
This pull request aims at creating a new base Docker image based on Alpine linux for all the Vimond microservices. 
The benefits of using Alpine linux as base layer are:
* Lightweight images
* Fail-fast containers
The latter benefit is crucial for settling the foundations of the new Experience VPC based on Kubernetes

The pull request tries to make the transition to this new base images as much transparent as possible. The current dockerfiles of the micros won't require any changes.

### CODE REVIEW

File: docker-service-alpine.sh [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    JAVA_NEW_RELIC_OPTIONS="-javaagent:/opt/$SERVICE_NAME/newrelic/newrelic.jar"
+fi
+
+java_exec="java $JAVA_OPTIONS $JAVA_NEW_RELIC_OPTIONS $JAVA_PROMETHEUS_AGENT_OPTIONS $JAVA_MEMORY $JAVA_LOGGC $JAVA_RUNTIME_OPTIONS -jar ${service_jar} $@  $SERVICE_CONFIG"
+
+if [ "$(whoami)" == "root" ] ; then
+    su $SERVICE_NAME -c "exec $java_exec"
+elif [ "$(whoami)" == ${SERVICE_NAME} ]; then
+    $java_exec
+else
```
Comment (dagvl) [inline docker-service-alpine.sh]: Do we need some kind of launcher here to deal with the PID 1 problem? In phusion, my_init dealt with it, is there anything that does it in apline by default? https://www.fpcomplete.com/blog/2016/10/docker-demons-pid1-orphans-zombies-signals is nice background reading and https://github.com/krallin/tini is an example (untested) minimal init implementation.

Comment (olvesh) [inline docker-service-alpine.sh]: We still keep phusion around. If your container spawns processes that should still be used for now.

**Comment by olvesh** [comment]
@matteo - you forgot to change circle.yml.  Maybe I said I'd do that, in that case I forgot :-)

I think this looks good. Add a circle.yml file using the makefile and merge (rebase). :+1:

**Comment by matteoluzzi** [comment]
I and @olvesh investigate on existing init scripts but none of them suit our use case. This is because they propagate signals only to the first child process and then exit. In our case the first process is bash which does not propagate SIGTERM signal to the java process. 
The easiest solution atm is to force to run the container as ${SERVICE_NAME} user in order to bypass the bash process and having the service process running as pid 1.

**Pull request merged**